### PR TITLE
Controlling code coverage externals version using Darc

### PR DESCRIPTION
**Pull request by jakubch1**
## Description
This PR is changing way how CC version is stored to enable automatic updates using Darc.

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Dependencies>
   <ProductDependencies>
+    <Dependency Name="Microsoft.Internal.CodeCoverage" Version="17.0.0-beta.21377.6">
```
Comment (Sanan07) [inline eng/Version.Details.xml:4]: Why don't you use `MicrosoftInternalCodeCoverageVersion` parameter from Versions.props here?

Comment (jakubch1) [reply eng/Version.Details.xml:4]: This version is controlled by Darc. This will be automatically changed.

Comment (jakubch1) [reply eng/Version.Details.xml:4]: Check here: https://github.com/dotnet/arcade/blob/main/Documentation/Darc.md#updating-dependencies-in-your-local-repository

Comment (jakubch1) [reply eng/Version.Details.xml:4]: This is example of such automatic update: https://devdiv.visualstudio.com/DevDiv/_git/vs-code-coverage/pullrequest/340431?_a=files&path=%2Feng%2FVersions.props
You can see that version is changed in both places

**Review by Haplois — APPROVED** [review]

**Pull request merged**